django/contrib/staticfiles/storage.py

```diff
@@ -154,44 +154,54 @@ def url_converter(self, name, template=None):
 
         def converter(matchobj):
             """
-            Converts the matched URL depending on the parent level (`..`)
-            and returns the normalized and hashed URL using the url method
-            of the storage.
+            Converts the matched URL to a normalized and hashed URL.
+
+            This requires figuring out which files the matched URL resolves
+            to and calling the url() method of the storage.
             """
             matched, url = matchobj.groups()
-            # Completely ignore http(s) prefixed URLs,
-            # fragments and data-uri URLs
-            if url.startswith(('#', 'http:', 'https:', 'data:', '//')):
+
+            # Ignore absolute/protocol-relative, fragments and data-uri URLs.
+            if url.startswith(('http:', 'https:', '//', '#', 'data:')):
+                return matched
+
+            # Ignore absolute URLs that don't point to a static file (dynamic
+            # CSS / JS?). Note that STATIC_URL cannot be empty.
+            if url.startswith('/') and not url.startswith(settings.STATIC_URL):
                 return matched
-            name_parts = name.split(os.sep)
-            # Using posix normpath here to remove duplicates
+
+            # This is technically not useful and could be considered a bug:
```

Comment:
@jezdez, do you have any input on this? (original commit: 1d32bdd3) I'd like to avoid such an odd comment.